### ISSUE TITLE
[BUG] approxConv is not updating ccw.measurement in calling function

### ISSUE THREAD

**Issue by dehann** [JuliaRobotics/IncrementalInference.jl#568]
```julia
fg = ...
fcm = solverData(getFactor(fg, :x1x2f1)).fnc.measurement |> deepcopy
approxConv(fg, :x1x2f1, :x1)
fcm2 = solverData(getFactor(fg, :x1x2f1)).fnc.measurement
```

and the values in fcm's are the same!  The factor `ccw.measurements` location should have been updated by the internal `evalFunctionSpecific` call.  Quick testing showed that the internal getSample method is called and `ccwl.measurements` is updated, but somehow that data is lost on the way out as `approxConv` finishes.

-----

These were somewhat useful during first finding this bug:
```julia
solverData(getFactor(fg, :x1x2f1)).fnc |> pointer_from_objref
solverData(getFactor(fg, :x1x2f1)).fnc.measurement[1] |> pointer_from_objref
```

**Comment by dehann** [comment]
also
```juila
  @show pointer_from_objref(ccw)
  @show (solverData(fct).fnc |> pointer_from_objref)
  @show solverData(fct).fnc.measurement[1] |> pointer_from_objref
```

**Comment by dehann** [comment]
Looks like this was resolved as part of DFG v0.6, IIF v0.9, and #639
